server/rest/src/main/java/org/infinispan/rest/resources/MetricsResource.java

```diff
@@ -52,9 +54,13 @@ private CompletionStage<RestResponse> metrics(RestRequest restRequest) {
          RestResponseBuilder<NettyRestResponse.Builder> builder = new NettyRestResponse.Builder();
 
          try {
+            PrometheusMeterRegistry registry = metricsCollector.registry();
+            if (registry == null) {
+               return builder.status(NOT_FOUND.code()).build();
```

Comment:
Will this not break other things? I wonder if we should instead require a PrometheusMeterRegistry impl instead of the MetricsRegistry interface.

Reply:
this is for safety and should never happen :) 
does quarkus support `PrometheusMeterRegistry` ? the examples I found online are using `MetricsRegistry`.